### PR TITLE
Adding anomaly detection sample domain to template manager

### PR DESCRIPTION
**Pull request by ashensw**

### CODE REVIEW

File: modules/distribution/repository/conf/template-manager/domain-template/SensorAnomalyDetectionDomain.xml [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+
+<domain name="SensorAnomalyDetection">
+    <description>Detecting anomalies in sensor reading</description>
+    <scenarios>
+        <scenario type="LinearRegressionWithLengthWindow">
+            <description>Anomalies in sensor reading will be detected using linear regression over a defined number of
+                events and those anomalies will be sent to the predefined alert stream
+            </description>
+            <templates>
+                <!--Note: These will be deployed in the order they appear here-->
```
Comment (dilini-muthumala) [inline modules/distribution/repository/conf/template-manager/domain-template/SensorAnomalyDetectionDomain.xml]: Shall we remove this comment, because this is not related to this sample? WDYT?

File: modules/distribution/repository/conf/template-manager/domain-template/SensorAnomalyDetectionDomain.xml [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+            insert into tempStream;
+        end;
+
+        from tempStream[notify == true]
+        select 'Anomaly' as type, str:concat("$sensorType sensor of sensor id ", sensorID, " had a value of ", sensorValue) as message, timeStamp as time, 'high' as severity
+        insert into OutputStream;
+                    ]]>
+                </template>
+
+                <!-- gadgets -->
```
Comment (dilini-muthumala) [inline modules/distribution/repository/conf/template-manager/domain-template/SensorAnomalyDetectionDomain.xml]: Also these comments...

File: modules/distribution/repository/conf/template-manager/domain-template/SensorAnomalyDetectionDomain.xml (hunk 1 above)
Comment (ashensw) [inline modules/distribution/repository/conf/template-manager/domain-template/SensorAnomalyDetectionDomain.xml]: sure, I will remove them.
